Sources/CasePaths/XCTUnwrap.swift

```diff
@@ -4,28 +4,33 @@ import Foundation
 /// Asserts that an enum value matches a particular case and returns the associated value.
 ///
 /// - Parameters:
-///   - expression: An enum value.
+///   - enum: An enum value.
 ///   - extract: A closure that attempts to extract a particular case from the enum. You can supply
 ///     a case path literal here, like '/Enum.case'.
 ///   - message: An optional description of a failure.
 /// - Returns: The unwrapped associated value from the matched case of the enum.
-public func XCTUnwrap<Root, Case>(
-  _ expression: @autoclosure () throws -> Root,
-  case extract: (Root) -> Case?,
+public func XCTUnwrap<Enum, Case>(
+  _ enum: @autoclosure () throws -> Enum,
+  case extract: (Enum) -> Case?,
   _ message: @autoclosure () -> String = "",
   file: StaticString = #file,
   line: UInt = #line
 ) throws -> Case {
-  guard let value = try extract(expression())
+  let `enum` = try `enum`()
+  guard let value = extract(`enum`)
   else {
     #if canImport(ObjectiveC)
       _ = XCTCurrentTestCase?.perform(Selector(("setContinueAfterFailure:")), with: false)
     #endif
     let message = message()
     XCTFail(
       """
-      XCTUnwrap failed: expected non-nil value of type "\(typeName(Case.self))"\
-      \(message.isEmpty ? "" : " - " + message)
+      XCTUnwrap failed: expected to extract value of type "\(typeName(Case.self))" from \
+      "\(typeName(Enum.self))"\
+      \(message.isEmpty ? "" : " - " + message) …
+
+        Actual:
+          \(String(describing: `enum`))
```

Comment:
I found `String(describing:)` to be good enough for most examples. Would be nice to use custom dump, but prob not a dependency we wanna bring in just for these helpers.